pytorch_lightning/trainer/training_loop.py

```diff
@@ -238,7 +232,7 @@ def on_train_batch_end(self, epoch_output, batch_end_outputs, batch, batch_idx,
         self.trainer.logger_connector.on_train_batch_end()
 
     def reset_train_val_dataloaders(self, model):
-        if not self.trainer.reload_dataloaders_every_epoch:
+        if self.trainer.train_dataloader is None or not self.trainer.reload_dataloaders_every_epoch:
```

Comment:
was it failing somewhere if we don't update it?

Reply:
For the first epoch, when `reload_dataloaders_every_epoch` is True, hence the `dataloaders` are not reset before the training loop.